pkg/analyzer/analyzers/jira/jira.go

```diff
@@ -0,0 +1,215 @@
+//go:generate generate_permissions permissions.yaml permissions.go jira
+
+package jira
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"slices"
+
+	"github.com/fatih/color"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+func (a Analyzer) Type() analyzers.AnalyzerType {
+	return analyzers.AnalyzerTypeJira
+}
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	token, exist := credInfo["token"]
+	if !exist {
+		return nil, fmt.Errorf("token not found in credential info")
+	}
+	domain, exist := credInfo["domain"]
+	if !exist {
+		return nil, fmt.Errorf("domain not found in credential info")
+	}
+	email, exist := credInfo["email"]
+	if !exist {
+		return nil, fmt.Errorf("email not found in credential info")
+	}
+
+	// analyze permissions
+	info, err := AnalyzePermissions(a.Cfg, token, domain, email)
+	if err != nil {
+		return nil, err
+	}
+
+	// secret info to analyzer
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+func AnalyzeAndPrintPermissions(cfg *config.Config, domain, email, token string) {
+	noInfo := true
+	info, err := AnalyzePermissions(cfg, token, domain, email)
+	if err != nil {
+		// just print the error in cli and continue as a partial success
+		color.Red("[x] Error : %s", err.Error())
+	} else {
+		color.Green("[!] Valid Jira API token\n\n")
+	}
+
+	if info == nil {
+		color.Red("[x] Error : %s", "No information found")
+		return
+	}
+
+	if info.UserInfo.AccountID != "" {
+		noInfo = false
+		printUserInfo(info.UserInfo)
+	}
+
+	if len(info.Permissions) > 0 {
+		noInfo = false
+		printPermissions(info.Permissions)
+	}
+
+	if len(info.Resources) > 0 {
+		noInfo = false
+		printResources(info.Resources)
+	}
+
+	if noInfo {
+		color.Yellow("[i] No information found for the provided token")
+	}
+}
+
+func AnalyzePermissions(cfg *config.Config, token, domain, email string) (*SecretInfo, error) {
+	// create http client
+	client := analyzers.NewAnalyzeClient(cfg)
+
+	var secretInfo = &SecretInfo{}
+
+	// capture the user information
+	statusCode, err := captureUserInfo(client, token, domain, email, secretInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	switch statusCode {
+	case 401:
+		return nil, fmt.Errorf("invalid email or api token")
+	case 404:
+		return nil, fmt.Errorf("domain not found: %s", domain)
+	}
+
+	body, _, err := checkAllJiraPermissions(client, domain, email, token)
```

Comment:
```suggestion
	body, _, err := capturePermissions(client, domain, email, token)
```

Would be better to keep uniformity in naming.